src/map_preview_graphic.c

```diff
@@ -0,0 +1,946 @@
+            SetBgPriority(GF_BG_LYR_MAIN_2, (u8) unk->bg2Prioirty);
+            SetBgPriority(GF_BG_LYR_MAIN_0, (u8) unk->bg0Prioirty);
+            SetBgPriority(GF_BG_LYR_MAIN_1, (u8) unk->bg1Prioirty);
+            SetBgPriority(GF_BG_LYR_MAIN_3, (u8) unk->bg3Prioirty);
+            reg_G2_BLDCNT = 0;
+            BG_ClearCharDataRange(2, 0x20, 0, unk->heapId);
+            BgClearTilemapBufferAndCommit(unk->bgConfig, 2);
+            WindowArray_Delete(unk->window, 1);
+            FreeToHeap(unk);
+            return TRUE;
+        }
+        break;
+    }
+    return FALSE;
+}
+
+static void MapPreviewGraphic_LoadGfx(UnkStruct_0206A388 *unk) {
+    GfGfxLoader_GXLoadPal(NARC_a_1_5_0, sMapPreviewGraphicData[unk->index].files[unk->timeIndex].palNo, GF_PAL_LOCATION_MAIN_BG, GF_PAL_SLOT_0_OFFSET, 0x160, unk->heapId);
+    GfGfxLoader_LoadCharData(NARC_a_1_5_0, sMapPreviewGraphicData[unk->index].files[unk->timeIndex].charNo, unk->bgConfig, GF_BG_LYR_MAIN_2, 0, 0, 1, unk->heapId);
+    GfGfxLoader_LoadScrnData(NARC_a_1_5_0, sMapPreviewGraphicData[unk->index].files[unk->timeIndex].scrnNo, unk->bgConfig, GF_BG_LYR_MAIN_2, 0, 0, 1, unk->heapId);
+}
+
+static void MapPreviewGraphic_InitWindow(UnkStruct_0206A388 *unk) {
+    unk->window = AllocWindows(unk->heapId, 1);
+    AddWindowParameterized(unk->bgConfig, unk->window, 2, 0, 0, 28, 2, 13, 0x2C0);
+}
+
+static int sub_0206A694(UnkStruct_0206A388 *unk, FieldSystem *fsys) {
+    LocalMapObject *player;
+    
+    if (!unk->unk30) {
+        return TRUE;
+    }
+
+    UnkStateMachineSubstruct_0206A388 *unkSub = &unk->unk34; //required to match
+    
+    switch (unk->unk34.state) {
+    case 0:
+        PlayerAvatar_ToggleAutomaticHeightUpdating_NowApply(fsys->playerAvatar, 0);
+        unkSub->state++;
+        break;
+    case 1:
+        VecFx32 pos;
+        player = PlayerAvatar_GetMapObject(fsys->playerAvatar);
+        unkSub->unk2++;
+        MapObject_GetPositionVec(player, &pos);
+        pos.y -= 2 * FX32_ONE;
+        MapObject_SetPositionVec(player, &pos);
+        if (unkSub->unk2 >= 16) {
+            PlayerAvatar_ToggleAutomaticHeightUpdating_NowApply(fsys->playerAvatar, TRUE);
+            unkSub->state++;
+        }
+        break;
+    case 2:
+        Field_PlayerAvatar_OrrTransitionFlags(fsys->playerAvatar, 1);
+        Field_PlayerAvatar_ApplyTransitionFlags(fsys->playerAvatar);
+        unkSub->state++;
+        break;
+    case 3:
+        player = PlayerAvatar_GetMapObject(fsys->playerAvatar);
+        if (MapObject_AreBitsSetForMovementScriptInit(player)) {
+            MapObject_SetHeldMovement(player, 1);
+            unkSub->state++;
+        }
+        break;
+    case 4:
+        player = PlayerAvatar_GetMapObject(fsys->playerAvatar);
+        if (MapObject_AreBitsSetForMovementScriptInit(player)) {
+            if (FollowMon_IsActive(fsys)) {
+                ov01_02205790(fsys, 1);
+                sub_0205FC94(FollowMon_GetMapObject(fsys), 0x30);
```

Comment:
0x30 is a movement constant; you can add it in, or don't

Reply:
Might as well add it in as an unk to be documented later